pixeltable/catalog/catalog.py

```diff
@@ -434,20 +455,23 @@ def _acquire_path_locks(
         - if lock_mutable_tree, also X-lock all mutable views of tbl
 
         Raises Error if tbl doesn't exist.
+        Return False if the lock couldn't be acquired (X-lock on a non-mutable table), True otherwise.
         """
         path_handles = tbl.get_tbl_versions()
         read_handles = path_handles[:0:-1] if for_write else path_handles[::-1]
         for handle in read_handles:
-            _ = self.get_tbl_version(handle.id, handle.effective_version)
+            _ = self.get_tbl_version(handle.id, handle.effective_version, validate_initialized=True)
         if not for_write:
-            return  # nothing left to lock
-        self._acquire_tbl_lock(
+            return True  # nothing left to lock
+        handle = self._acquire_tbl_lock(
             tbl_id=tbl.tbl_id,
             for_write=True,
             lock_mutable_tree=lock_mutable_tree,
             raise_if_not_exists=True,
             check_pending_ops=check_pending_ops,
         )
+        _ = self.get_tbl_version(path_handles[0].id, path_handles[0].effective_version, validate_initialized=True)
```

Comment:
This is counterintuitive. Are we relying on `get_tbl_version()` having side effects here?

Reply:
It updates the cache. Left comment.

Reply:
Ok, I wonder if it'd be clearer to have a separate `_update_tbl_version_cache()` that is called by `get_tbl_version()` and can also be called separately when needed.

It always makes me squint a bit when a `get_*()` is called for its side effects.